src/Avalonia.X11/Glx/GlxDisplay.cs

```diff
@@ -126,7 +126,11 @@ private IntPtr CreatePBuffer()
             GlxContext Create(GlVersion profile)
             {
                 var profileMask = GLX_CONTEXT_CORE_PROFILE_BIT_ARB;
-                if (profile.Type == GlProfileType.OpenGLES) 
+                if (profile.Type == GlProfileType.OpenGL && 
+                    profile.EnableCompatibilityProfile && 
+                    profile.Major * 10 + profile.Minor >= 32)
```

Comment:
That's not a proper way to compare versions. Consider using System.Version struct if you don't want to implement the logic manually.